Scripts/RemediationScripts/Remediate-EnableHttpsForStorageAccount.ps1

```diff
@@ -0,0 +1,723 @@
+        $SubscriptionId,
+
+        [switch]
+        [Parameter(Mandatory = $false, HelpMessage="Specifies a forceful remediation without any prompts")]
+        $Force,
+
+        
+        [string]
+        [Parameter(Mandatory = $true, HelpMessage="CSV file path which contain logs generated by remediation script to rollback remediation changes")]
+        $FilePath
+    )
+
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "Starting rollback operation to disable 'EnableHttpsTrafficOnly' on Storage Account(s) from subscription [$($SubscriptionId)]...." -ForegroundColor $([Constants]::MessageType.Info)
+    write-host "$([Constants]::DoubleDashLine)"
+
+    try 
+    {   Write-Host "[step 1 of 3 ] : Checking pre-requisites and validiating the user."
+        Write-Host "Checking for pre-requisites..." 
+        Setup-Prerequisites
+        Write-Host $([Constants]::SingleDashLine)    
+    }     
+    catch 
+    {
+        Write-Host "Error occurred while checking pre-requisites. ErrorMessage [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)    
+        break
+    } 
+         
+    # Connect to AzAccount
+    $isContextSet = Get-AzContext
+    if ([string]::IsNullOrEmpty($isContextSet))
+    {       
+        Write-Host "Connecting to AzAccount..."
+        Connect-AzAccount -ErrorAction Stop
+        Write-Host "Connected to AzAccount" -ForegroundColor $([Constants]::MessageType.Update)
+    }
+
+    # Setting context for current subscription.
+    $currentSub = Set-AzContext -SubscriptionId $SubscriptionId -ErrorAction Stop -Force
+    Write-Host "Metadata Details: `nSubscriptionId: $($SubscriptionId) `nAccountName: $($currentSub.Account.Id) `nAccountType: $($currentSub.Account.Type)" 
+   
+    Write-Host $([Constants]::SingleDashLine) 
+    Write-Host "Starting with subscription [$($SubscriptionId)]..." 
+
+    Write-Host " ***To perform rollback operation for enabling 'EnableHttpsTrafficonly' user must have atleast contributor access on Storage Account(s) of subscription: [$($SubscriptionId)]  ***" -ForegroundColor $([Constants]::MessageType.Info)
+
+    Write-Host "Validating whether the current user [$($currentSub.Account.Id)] have valid account type [User] to run the script for subscription [$($SubscriptionId)]..." 
+
+    # Safe Check: Checking whether the current account is of type User
+    
+    if($currentSub.Account.Type -ne "User")
+    {
+        Write-Host "Warning: This script can only be run by user account type." -ForegroundColor $([Constants]::MessageType.Warning)
+        break;
+    }
+
+    Write-Host "Successfully validated" -ForegroundColor $([Constants]::MessageType.Update)
+    Write-Host $([Constants]::SingleDashLine) 
+    Write-Host "Fetching remediation log to perform rollback operation on  Storage Account(s) from subscription [$($SubscriptionId)]..." 
+
+    # Array to store resource context
+    $resourceContext = @()
```

Comment:
Reference to `Context` can be misleading. Please rename this variable.

Reply:
Done